### PR TITLE
gh-92063:  Enforce types in specialized PRECALL opcodes

### PR DESCRIPTION
**Pull request by sweeneyde**
https://github.com/python/cpython/issues/92063

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2022-04-30-04-26-01.gh-issue-92063.vHnhf6.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+The ``PRECALL_METHOD_DESCRIPTOR_FAST_WITH_KEYWORDS`` now ensures methods are called only on objects of the correct type.
```
Comment (sweeneyde) [inline Misc/NEWS.d/next/Core and Builtins/2022-04-30-04-26-01.gh-issue-92063.vHnhf6.rst]:
```suggestion
The ``PRECALL_METHOD_DESCRIPTOR_FAST_WITH_KEYWORDS`` instruction
now ensures methods are called only on objects of the correct type.
```

**Comment by sweeneyde** [comment]
The single failing buildbot fails on repeated invocations of `test_embed.test_finalize_structseq`, which is related to https://github.com/python/cpython/issues/92031

**Comment by markshannon** [comment]
@sweeneyde Thanks for fixing this.
Looks good to me.

File: Misc/NEWS.d/next/Core and Builtins/2022-04-30-04-26-01.gh-issue-92063.vHnhf6.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The ``PRECALL_METHOD_DESCRIPTOR_FAST_WITH_KEYWORDS`` instruction
+now ensures methods are called only on objects of the correct type.
```
Comment (sweeneyde) [inline Misc/NEWS.d/next/Core and Builtins/2022-04-30-04-26-01.gh-issue-92063.vHnhf6.rst:2]:
```suggestion
Specialized forms of the ``PRECALL`` instruction
now ensure methods are called only on objects of the correct type.
```

**Pull request merged**